[macOS] v. 5.9.7 broken on macOS < 10.13: `if_dl.h: error: unknown type name 'u_char'; did you mean 'char'?`

5.9.7 fails to build now on multiple systems.

On buildbots with clang, `x86_64`:
```
/usr/bin/clang -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -pipe -Os -arch x86_64 -isysroot/ -DPSUTIL_POSIX=1 -DPSUTIL_SIZEOF_PID_T=4 -DPSUTIL_VERSION=597 -DPy_LIMITED_API=0x03060000 -DPSUTIL_OSX=1 -I/opt/local/Library/Frameworks/Python.framework/Versions/3.11/include/python3.11 -c psutil/arch/osx/net.c -o build/temp.macosx-10.12-x86_64-cpython-311/psutil/arch/osx/net.o
In file included from psutil/arch/osx/net.c:12:
/usr/include/net/if_dl.h:90:2: error: unknown type name 'u_char'; did you mean 'char'?
        u_char  sdl_len;        /* Total length of sockaddr */
        ^
/usr/include/net/if_dl.h:91:2: error: unknown type name 'u_char'; did you mean 'char'?
        u_char  sdl_family;     /* AF_LINK */
        ^
/usr/include/net/if_dl.h:92:2: error: unknown type name 'u_short'; did you mean 'short'?
        u_short sdl_index;      /* if != 0, system given index for interface */
        ^
/usr/include/net/if_dl.h:93:2: error: unknown type name 'u_char'; did you mean 'char'?
        u_char  sdl_type;       /* interface type */
        ^
/usr/include/net/if_dl.h:94:2: error: unknown type name 'u_char'; did you mean 'char'?
        u_char  sdl_nlen;       /* interface name length, no trailing 0 reqd. */
        ^
/usr/include/net/if_dl.h:95:2: error: unknown type name 'u_char'; did you mean 'char'?
        u_char  sdl_alen;       /* link level address length */
        ^
/usr/include/net/if_dl.h:96:2: error: unknown type name 'u_char'; did you mean 'char'?
        u_char  sdl_slen;       /* link layer selector length */
        ^
psutil/arch/osx/net.c:68:21: warning: array subscript is of type 'char' [-Wchar-subscripts]
            ifc_name[sdl->sdl_nlen] = 0;
                    ^~~~~~~~~~~~~~
1 warning and 7 errors generated.
XCode (https://developer.apple.com/xcode/) is not installed
error: command '/usr/bin/clang' failed with exit code 1
```

Locally with gcc, on `ppc`:
```
/usr/bin/gcc-4.2 -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -pipe -Os -arch ppc -isysroot/ -DPSUTIL_POSIX=1 -DPSUTIL_SIZEOF_PID_T=4 -DPSUTIL_VERSION=597 -DPy_LIMITED_API=0x03060000 -DPSUTIL_OSX=1 -I/opt/local/Library/Frameworks/Python.framework/Versions/3.11/include/python3.11 -c psutil/arch/osx/net.c -o build/temp.macosx-10.6-ppc-cpython-311/psutil/arch/osx/net.o
In file included from psutil/arch/osx/net.c:12:
//usr/include/net/if_dl.h:90: error: expected specifier-qualifier-list before ‘u_char’
psutil/arch/osx/net.c: In function ‘psutil_net_io_counters’:
psutil/arch/osx/net.c:67: error: ‘struct sockaddr_dl’ has no member named ‘sdl_data’
psutil/arch/osx/net.c:67: error: ‘struct sockaddr_dl’ has no member named ‘sdl_nlen’
psutil/arch/osx/net.c:67: error: ‘struct sockaddr_dl’ has no member named ‘sdl_data’
psutil/arch/osx/net.c:67: error: ‘struct sockaddr_dl’ has no member named ‘sdl_nlen’
psutil/arch/osx/net.c:68: error: ‘struct sockaddr_dl’ has no member named ‘sdl_nlen’
XCode (https://developer.apple.com/xcode/) is not installed
error: command '/usr/bin/gcc-4.2' failed with exit code 1

ERROR Backend subprocess exited when trying to invoke build_wheel
```

It does something strange. First of all, Xcode is installed. Then, why it needs Xcode at all?
Finally, something got badly broken with headers, apparently. 5.9.5 built fine.

--- a/psutil/arch/osx/net.c
+++ b/psutil/arch/osx/net.c
@@ -9,11 +9,11 @@
 // https://github.com/giampaolo/psutil/blame/efd7ed3/psutil/_psutil_osx.c
 
 #include <Python.h>
+#include <net/if.h>
 #include <net/if_dl.h>
 #include <net/route.h>
 #include <sys/sysctl.h>
 #include <sys/socket.h>
-#include <net/if.h>
 
 #include "../../_psutil_common.h"
 

